### ISSUE TITLE
import file without extension suffix

### ISSUE THREAD

**Issue by chengjianhua** [kentcdodds/import-all.macro#24]
- `import-all.macro` version:
- `node` version:
- `npm` (or `yarn`) version:

Relevant code or config

```javascript
import importAll from 'import-all.macro'

const entityClassesMap = importAll.sync('../../entities/*.*')
```

What you did:

What happened:

Output:
```javascript
var _entitiesAppRepositoryTs = _interopRequireWildcard(require("../../entities/app-repository.ts"));

var _entitiesAppTs = _interopRequireWildcard(require("../../entities/app.ts"));
```

Reproduction repository:

Problem description:

I use typescript to code my project, so the files under `../../entities/*.*` directory are all with extension `.ts`, `importAll.sync()` also generated imports for full paths with suffix `.ts` too.

The `babel-node` can execute the codes generated by `import-all.marco`. But after those `.ts` files were compiled into `.js` files, we can't execute the `.js` file contain `require` for  `.ts` files. Because there are not `.ts` files in dist directory.

Suggested solution:

Usually, we import js or ts modules without extension, so our import or requirement will work very well with the "same" file with different extensions.

So maybe `import-all.macro` can provide an option to strip the extension suffix when generated import statements?

The configuration in babel-plugin-macro.config.js

```javascript
  importAll: {
    ignoreExtensions: ['.js', '.ts']
  }
```

**Comment by chengjianhua** [comment]
@kentcdodds PTAL~ 🙆‍♂️

**Comment by kentcdodds** [comment]
I have an idea for this. How about we give you this API:

```js
importAll: {
  transformModulePath(modulePath, importingModule) {
    return newModulePath
  },
}
```

This would allow for all kinds of flexibility.

**Comment by kentcdodds** [comment]
If that looks good to you, then feel free to implement it in [a pull requests](http://makeapullrequest.com) (with tests please).

**Comment by chengjianhua** [comment]
> I have an idea for this. How about we give you this API:
> 
> ```js
> importAll: {
>   transformModulePath(modulePath, importingModule) {
>     return newModulePath
>   },
> }
> ```
> 
> This would allow for all kinds of flexibility.

This looks good for me~ I'm glad to implement it with tests.

**Comment by chengjianhua** [comment]
@kentcdodds I have submitted my PR #25. PTAL again~

**Comment by kentcdodds** [comment]
Thanks @chengjianhua. This is fixed in #24
